### ISSUE TITLE
How to determine type of leaving picture in picture mode ?

### ISSUE THREAD

**Issue by kmsheng** [w3c/picture-in-picture#228]
![CleanShot 2024-07-05 at 16 49 42@2x](https://github.com/w3c/picture-in-picture/assets/880569/f19fc1fb-b7b6-47e4-bc3a-bc2cd6f5c2ce)

Is there any way to determine what button was triggered in `leavepictureinpicture ` event ?

**Comment by beaufortfrancois** [comment]
It's not possible as far as I know. Out of curiosity, what is your case?

**Comment by kmsheng** [comment]
@beaufortfrancois I have a streamer page that has a video element, once I open the picture-in-picture window and switch to other page. I need to go back to the streamer page by clicking the `Back to tab` button.

**Comment by kmsheng** [comment]
There's a [workaround](https://stackoverflow.com/questions/55991590/how-to-determine-type-of-leaving-picture-in-picture-mode) to solve this issue, but it fails for the paused video.

**Comment by beaufortfrancois** [comment]
Sorry, it's not clear to me what's the use case yet.
If there were a reason string in `onleavepictureinpicture` for instance, what would it allow you to do?

```js
video.onleavepictureinpicture = (event) => {
  if (event.reason == "backToTab") {
    // What would you differently?
  } else {
    // Do what you do.
  }
};
```

**Comment by kmsheng** [comment]
@beaufortfrancois Hey, appreciate your response. Then it would allow me to go back to the original streamer page by clicking the “Back to tab” button of pip video. It’s not really a full page redirect, it’s a frontend route change triggered by vue router. ( I hide my video element in other page, but it’s still on DOM tree )

**Comment by kmsheng** [comment]
@beaufortfrancois I don’t know if my use case is common, most of the demos I see online that “back to tab” usually happens on the same page.

**Comment by beaufortfrancois** [comment]
I wonder if using the [Page Visibility API](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API) would be enough in your use case to know when the tab is visible again. What do you think?

**Comment by kmsheng** [comment]
Probably not, because it’s on the same tab.

**Comment by kmsheng** [comment]
YouTube has a similar approach, but it’s done with a regular fixed position DOM element ( not using pip api )

**Comment by beaufortfrancois** [comment]
You may want to use the Document Picture-in-Picture API. Look at https://developer.chrome.com/docs/web-platform/document-picture-in-picture/

**Comment by kmsheng** [comment]
Yes, documentPictureInPicture is a desireable solution if it has a better browser support rate.

**Comment by kmsheng** [comment]
I've created a simple demo video to illustrate the use case.
https://www.youtube.com/watch?v=bNuw1CpZSKY

**Comment by kmsheng** [comment]
In above video, when I click the close button, I want to stay in the about page. 
Currently there's no way to distinguish between close btn and `back to tab` button.
Reason string would be a good fit in this example.

**Comment by beaufortfrancois** [comment]
We'll monitor how many people will vote for this issue then and we'll see if adding a reason string should be added to the `onleavepictureinpicture` event listener.

**Comment by kmsheng** [comment]
Sure, thank you for your support : )
